lib/internal/Magento/Framework/Mview/View/Subscription.php

```diff
@@ -250,10 +255,26 @@ protected function buildStatement($event, $changelog)
             $trigger,
             $this->connection->quoteIdentifier($this->resource->getTableName($changelog->getName())),
             $this->connection->quoteIdentifier($changelog->getColumnName()),
-            $this->connection->quoteIdentifier($this->getColumnName())
+            $this->connection->quoteIdentifier($column)
         );
     }
 
+    /**
+     * Returns subscription column name by view
+     *
+     * @param ViewInterface $view
+     * @return string
+     */
+    private function getSubscriptionColumn(ViewInterface $view): string
+    {
+        $subscriptions = $view->getSubscriptions();
+        if (!isset($subscriptions[$this->getTableName()]['column'])) {
+            throw new \RuntimeException(sprintf('Column name for view with id "%s" doesn\'t exist', $view->getId()));
```

Comment:
We probably should catch this exception somewhere, as it is I guess it would return an error to the user and not useful message, we should add a proper error message for the user (wherever we catch this exception), and probably log a more descriptive error on logs.

Reply:
Hi @gabrieldagama, the exception is already being handled here
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Indexer/Controller/Adminhtml/Indexer/MassChangelog.php#L40
Could you please take a look?
Thank you.